### ISSUE TITLE
Classification: Close the feedback loop — local bias from thumbs up/down

### ISSUE THREAD

**Issue by EldestGruff** [EldestGruff/PersonalAI#48]
## Overview
The thumbs up/down feedback UI on the detail screen is fully wired — it stores feedback signals in CoreData via `FineTuningService`. But the loop is not closed: that feedback never influences future classification decisions. This issue implements Option A (local bias layer) as a meaningful first step, designed to be fully compatible with Option B (model fine-tuning) later.

## Design Principle: Compatibility with Future Fine-Tuning (Option B)
The raw feedback data stored now (`FineTuningData` CoreData records with reward signals, positive/negative flags, corrections) is exactly what a future model fine-tuning pipeline will need. Option A adds a **local bias layer on top of** that stored data — it does not replace or alter it. When Option B arrives:
- The CoreData feedback records feed the model directly
- The local bias table is simply retired
- No data is lost in the transition

## What Option A Does
When a user marks a classification as unhelpful (👎), the system learns a local correction preference. On future classifications of similar thoughts, the correction is applied as a weighted nudge toward the preferred type.

Example:
- User captures "pick up dry cleaning" → classified as `idea`
- User marks it 👎
- Next time a similar short action-oriented thought is captured, `task` or `reminder` gets a bias bump

This does not require retraining any model. It is a lightweight rule layer applied post-classification.

### 1. `ClassificationBiasStore` (new, lightweight)
A `UserDefaults`-backed store (or small CoreData entity) that maintains a correction table:

```swift
struct ClassificationCorrection: Codable {
    let pattern: String        // lowercased first 5 words of thought content
    let originalType: String   // what the model guessed
    let preferredType: String  // what the user corrected to (or signaled via feedback)
    let weight: Double         // increases with repeated negative feedback on same pattern
    let createdAt: Date
}
```

### 2. Wire negative feedback to bias store
In `FineTuningService.trackUserFeedback()`, when `isPositive == false`:
- Extract a pattern key from the thought content (first 5 words, lowercased)
- If the current classification type is known, store a correction entry
- Increment weight on repeated corrections of the same pattern

### 3. Apply bias in `ClassificationService`
After the model returns a classification result, check `ClassificationBiasStore` for any corrections matching the pattern:
- If a correction exists with sufficient weight (threshold: 2+ negative signals), nudge the confidence score of the preferred type upward
- If the bias-adjusted preferred type exceeds the model's result confidence, swap the classification type (keep original confidence score)
- Log that a bias correction was applied (for future debugging/Option B handoff)

### 4. Positive feedback reinforcement
When `isPositive == true`:
- If a bias correction exists for this pattern, reduce its weight (the model was right this time)
- If weight drops to zero, remove the correction entry

### 5. Bias store maintenance
- Cap at 200 correction entries (drop oldest when over limit)
- Entries older than 90 days decay in weight automatically
- Accessible in a future "Classification Preferences" settings section

## Compatibility Notes
- `FineTuningData` CoreData records are untouched — still accumulating for Option B
- Bias corrections are a separate, lightweight store
- When Option B is implemented: bias store can be exported as additional training signal, then cleared
- No API calls, no cloud dependency, works fully offline

## Acceptance Criteria
- [ ] Thumbs down on a thought stores a correction entry in `ClassificationBiasStore`
- [ ] A second capture with similar content gets the corrected type applied
- [ ] Thumbs up reduces correction weight for that pattern
- [ ] Bias corrections do not affect stored `FineTuningData` records
- [ ] Store is capped and self-maintaining
- [ ] No visible change to the user — feedback UI stays the same, classification just improves silently

## Implementation Order
Depends on: classification system (existing ✅), feedback UI (existing ✅), FineTuningService (existing ✅)
Can be implemented independently of all other open issues.
